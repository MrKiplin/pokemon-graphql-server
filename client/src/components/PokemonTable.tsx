import * as React from "react";
import { Table } from "semantic-ui-react";
import { PokemonQuery } from "../__generated__/graphql";

interface PokemonTableProps {
  pokemonQuery: PokemonQuery;
}

const PokemonTable: React.FC<PokemonTableProps> = ({ pokemonQuery }) => (
  <Table celled>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell>ID</Table.HeaderCell>
        <Table.HeaderCell>Name</Table.HeaderCell>
        <Table.HeaderCell>Types</Table.HeaderCell>
      </Table.Row>
    </Table.Header>

    <Table.Body>
      <Table.Row>
        <Table.Cell>{pokemonQuery.pokemon.id}</Table.Cell>
        <Table.Cell>{pokemonQuery.pokemon.name}</Table.Cell>
        <Table.Cell>{pokemonQuery.pokemon.types}</Table.Cell>
      </Table.Row>
    </Table.Body>
  </Table>
);

export default PokemonTable;
