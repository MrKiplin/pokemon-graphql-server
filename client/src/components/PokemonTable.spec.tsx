import { mount } from "enzyme";
import PokemonTable from "./PokemonTable";

describe("PokemonTable", () => {
  it("Should match props and snapshot", () => {
    const component = mount(
      <PokemonTable
        {...{
          pokemonQuery: {
            pokemon: {
              __typename: "PokemonGQL",
              id: "test-pokemon-id",
              name: "test-pokemon-name",
              types: ["test-pokemon-type"],
              imageURL: "test-image-url",
            },
          },
        }}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.find("PokemonTable").prop("pokemonQuery")).toEqual({
      pokemon: {
        __typename: "PokemonGQL",
        id: "test-pokemon-id",
        name: "test-pokemon-name",
        types: ["test-pokemon-type"],
        imageURL: "test-image-url",
      },
    });
  });
});
