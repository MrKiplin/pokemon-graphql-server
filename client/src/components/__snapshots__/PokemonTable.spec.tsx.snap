// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonTable Should match props and snapshot 1`] = `ReactWrapper {}`;
